libraries/WiFi/src/WiFiSTA.cpp

```diff
@@ -683,7 +683,9 @@ void WiFiSTAClass::_smartConfigCallback(uint32_t st, void* result) {
     smartconfig_status_t status = (smartconfig_status_t) st;
     log_d("Status: %s", sc_status_strings[st % 5]);
     if (status == SC_STATUS_GETTING_SSID_PSWD) {
+#if ARDUHAL_LOG_LEVEL >= ARDUHAL_LOG_LEVEL_DEBUG
         smartconfig_type_t * type = (smartconfig_type_t *)result;
```

Comment:
you should also include the call to log_d in the `#if`. Same for the one below. Makes the reason for the `#if` clearer.